### DISCUSSION TITLE
Verbose flag for HTML generation

### DISCUSSION THREAD

**Discussion by pelikhan** [gatsbyjs/gatsby#31894] [Help]
After running on some Javascript out of memory issues, I followed the nice guide on diagnosing and fixing those issues (thank you for the help). While running process-top, I could see the working set jump from 240Mb to 3.4Gb repeatidely. Is there a way to turn on some kind of uber-logging more in order to pinpoint which HTML page is generating some much memory usage. This would definitely diagnosing this further.

**Comment by KyleAMathews** [comment]
This plugin wraps Node's inspector module — if you open the generated `heap` file in Chrome after a build, you can see memory allocations https://www.gatsbyjs.com/plugins/gatsby-plugin-inspector/

Note: If I recall correctly, it doesn't capture activity on child processes (SSR happens in jest-worker worker processes). To get around that, it'd take some hacking but you could change the code [here](https://github.com/gatsbyjs/gatsby/blob/c87d1d116e33eb693f65bcfc57cc63b072fe4a5c/packages/gatsby/src/commands/build-html.ts#L221-L250) to _not_ use the worker pool but instead run SSR in the parent process. That way you could see what code is doing the allocation.

Another option too is to turn off all the routes and add them back one by one to see which one is causing trouble.

**Reply by pelikhan** [reply]
Thanks!
